### PR TITLE
Add new parser for Trivy

### PR DESCRIPTION
**Pull request by tofuatjava**
be aware that a new analyser-version need to be used

### CODE REVIEW

**Comment by tofuatjava** [comment]
how is the correct way to reference to an unreleased dependency of analysis-model?

**Comment by uhafner** [comment]
You need to rebase your analysis-model PR with `master`, see error at the end of https://ci.jenkins.io/job/Plugins/job/analysis-model/job/PR-542/13/console.

Then we get an incremental version. This version can be used in a PR of the associated https://github.com/jenkinsci/analysis-model-api-plugin. 

Then you can use this version in your PR.

**Comment by uhafner** [comment]
You can now use the released version:

https://github.com/jenkinsci/analysis-model/releases/tag/v9.5.0

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public static class Descriptor extends ReportScanningToolDescriptor {
+
+        /** Creates the descriptor instance. */
+        public Descriptor() {
+            super("trivy-vulnerabilityscanning");
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "aquasec trivy vulnerability scanner";
```
Comment (uhafner) [inline plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java:47]: This is a short label for the URL, what about Trivy?

It should be part of the file Messages.properties.

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        public String getUrl() {
+            return "https://github.com/aquasecurity/trivy";
+        }
+
+        @Override
+        public boolean isPostProcessingEnabled() {
+            return false;
+        }
+    }
+}
```
Comment (uhafner) [inline plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java:72]: The next time it would be helpful if you can create the PR in a branch of your fork. Otherwise the PR contains a lot of unrelated changes, hard to review.

**Review by uhafner — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @Override
+    public IssueParser createParser() {
+        return new TrivyParser();
+    }
+
+    /** Descriptor for this static analysis tool. */
+    @Extension
+    public static class Descriptor extends ReportScanningToolDescriptor {
```
[the remark this answers is not in the document]
Comment (tofuatjava) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java:38]: trivy is available under Apache Common 2.0 license. I do not know if the logo has an additional license or trademark.

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        @Override
+        public String getUrl() {
+            return "https://github.com/aquasecurity/trivy";
+        }
+
+        @Override
+        public boolean isPostProcessingEnabled() {
+            return false;
```
[the remark this answers is not in the document]
Comment (tofuatjava) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java:69]: there are no affected files, because the scan is done in a temporary image based on the original image. however, in my opinion it makes no sense to copy affected files anywhere.

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import edu.hm.hafner.analysis.parser.TrivyParser;
+import hudson.Extension;
+import io.jenkins.plugins.analysis.core.model.ReportScanningTool;
+
+/**
+ * <p>
+ * aquasec Trivy is a container vulnerability scanner.
+ * </p>
+ * <p>
+ * <strong>Usage: </strong>trivy image -f json -o results.json
```
[the remark this answers is not in the document]
Comment (tofuatjava) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java:15]: wilco

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java (hunk 2 above)
Comment (tofuatjava) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java:72]: wilco

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (tofuatjava) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Trivy.java:38]: have resized the logo to 48x48 and 24x24. I'm not sure if it is too pixelated now

File: plugin/src/main/resources/io/jenkins/plugins/analysis/warnings/Messages.properties [hunk 6]
```diff
@@ -127,6 +127,7 @@ Warnings.RFLint.ParserName=Robot Framework Lint
 Warnings.LinuxKernelOutput.ParserName=Linux Kernel Output Parser
 Warnings.WarningsPlugin.ParserName=Warnings Plugin Native Format
 Warnings.ProtoLint.ParserName=ProtoLint
+Warnings.Trivy.ParserName=aquasec trivy vulnerability scanner
```
Comment (uhafner) [inline plugin/src/main/resources/io/jenkins/plugins/analysis/warnings/Messages.properties:130]: Labels should be capitalized.

I still think that the text in Jenkins side panel would look better if you use *Trivy* rather than *Aquasec Trivy Vulnerability Scanner*.

**Review by uhafner — CHANGES_REQUESTED** [review]
Can you please add a line to `icons\License.txt` that adds the license for the icon.

And please rerun the test `ToolsLister` so that the supported tools file will reflect the new changes.

**Review by uhafner — APPROVED** [review]
